src/naip_cnn/acquisitions.py

```diff
@@ -0,0 +1,84 @@
+import ee
+
+from naip_cnn.config import CRS
+
+
+class Acquisition:
+    """A LiDAR aquisition with associated NAIP imagery in Earth Engine."""
+
+    def __init__(self, name: str, start_date: str, end_date: str) -> None:
+        self.name = name
+        self.start_date = start_date
+        self.end_date = end_date
+
+    @property
+    def proj(self):
+        return ee.Projection(CRS)
+
+    @property
+    def geometry(self):
+        return ee.Image(
+            f"projects/ee-maptheforests/assets/malheur_lidar/{self.name}"
+        ).geometry()
+
+    @property
+    def mask(self):
+        """A mask of fast loss over the acqusition period."""
+        return (
+            ee.ImageCollection("USFS/GTAC/LCMS/v2022-8")
+            .filter(ee.Filter.eq("study_area", "CONUS"))
+            .filterDate(self.start_date, self.end_date)
+            .select("Change")
+            .map(lambda img: img.eq(3))
+            .max()
+            .eq(0)
+            .reproject(self.proj.atScale(30))
+        )
+
+    def load_naip(self):
+        return (
+            ee.ImageCollection("USDA/NAIP/DOQQ")
+            .filterDate(self.start_date, self.end_date)
+            .filterBounds(self.geometry.bounds())
+            .mosaic()
+            .updateMask(self.mask)
+            .reproject(self.proj.atScale(1))
+        )
+
+    def load_lidar(self):
+        return (
+            ee.Image(f"projects/ee-maptheforests/assets/malheur_lidar/{self.name}")
+            .updateMask(self.mask)
+            .reproject(self.proj.atScale(30))
+        )
+
+
+# OR NAIP years: 2004 (RGB), 2005 (RGB), 2009, 2011, 2012, 2014, 2016, 2020, 2022
```

Comment:
More a question than anything.  I'm wondering if you had a rule for associating lidar acquisitions with NAIP acquisitions in terms of date.  For example, the 2007 lidar uses the 2009 NAIP (closest with 4-band data).  It looks like most of the time, you tried to use a later NAIP acquisition, but only if it was closer temporally (e.g. the 2017 lidar acquisitions use 2016 NAIP).  With the LCMS masking in there, this is definitely less of an issue.

Reply:
Good question, as there's definitely different ways to approach this! As you guessed, I prioritized the closest acquisition in time, and went with the more recent acquisition in the case of a tie. For 2007 that was done to avoid the RGB-only imagery from 2005, but generally I figured that more recent imagery should *theoretically* be higher quality. As you mentioned, masking disturbances should hopefully remove any changes that occur within a couple years, but maybe we should ultimately limit our validation to higher-confidence acquisitions that occurred in the same year?

Another idea I haven't implemented would be to train with multiple NAIP years for each LiDAR acquisition, e.g. a single canopy cover chip from 2018 could be compared against NAIP imagery from both 2016 and 2020, effectively doubling our training data. Given the amount of training data we already have to work with, I'm not sure this is really worth pursuing, but it's something to consider. We would need to be careful not to train with one year and validate with another over the same plot, though...

Reply:
> Another idea I haven't implemented would be to train with multiple NAIP years for each LiDAR acquisition, e.g. a single canopy cover chip from 2018 could be compared against NAIP imagery from both 2016 and 2020, effectively doubling our training data. Given the amount of training data we already have to work with, I'm not sure this is really worth pursuing, but it's something to consider.

I completely defer to you on this, but this seems like it would add some complexities for not a lot of gain.  Given that you're starting to think about how to minimize differences in NAIP collections (#13, #14), that might even serve to produce more-or-less duplicate training data.  But there might be an upside that I haven't considered?

To be clear, I think your current approach is great and preferentially choosing a later NAIP acquisition makes sense to me.

Reply:
I suspect that even with those normalization steps, there will still be enough difference between a footprint in subsequent NAIP acquisitions (shadow angle, radial distortion, etc.) that it might help the model to generalize further, but I agree that we should probably avoid adding that complexity for now. We've got enough things to worry about already!